xgi/classes/reportviews.py

```diff
@@ -268,10 +277,19 @@ def __getitem__(self, id):
         float
             The degree of an ID, weighted or unweighted
         """
-        weight = self._weight
-        if weight is None:
-            return len(self._ids[id])
-        return sum(self._id_attrs[dd].get(weight, 1) for dd in self._ids(id))
+
+        try:
+            return self._deg[id_bunch]
+        except:
```

Comment:
```suggestion
        except KeyError:
```

Reply:
Actually, I needed to add two different types of exceptions. Should work now.